src/main/java/com/ikubinfo/service/CabinService.java

```diff
@@ -0,0 +1,109 @@
+package com.ikubinfo.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.validation.ValidationException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.ikubinfo.converter.CabinConverter;
+import com.ikubinfo.dto.AttributeDto;
+import com.ikubinfo.dto.CabinDto;
+import com.ikubinfo.entities.AttributeEntity;
+import com.ikubinfo.entities.CabinEntity;
+import com.ikubinfo.entities.SiteEntity;
+import com.ikubinfo.enums.AttributeType;
+import com.ikubinfo.enums.BadRequestMessage;
+import com.ikubinfo.enums.ExceptionMessage;
+import com.ikubinfo.enums.ValidationMessage;
+import com.ikubinfo.exceptions.BadRequestException;
+import com.ikubinfo.exceptions.NotFoundException;
+import com.ikubinfo.repository.AttributeRepository;
+import com.ikubinfo.repository.CabinRepository;
+import com.ikubinfo.repository.SiteRepository;
+
+@Service
+@Transactional
+public class CabinService {
+
+	@Autowired
+	private CabinRepository cabinRepository;
+
+	@Autowired
+	private CabinConverter cabinConverter;
+
+	@Autowired
+	private SiteRepository siteRepository;
+
+	@Autowired
+	private AttributeRepository attributeRepository;
+
+	public CabinDto createCabin(CabinDto cabinToBeCreated) {
+		if (cabinToBeCreated.getId() != null) {
+			throw new ValidationException(ValidationMessage.DATA_NOT_VALID.getMessage());
+		}
+		SiteEntity site = siteRepository.findOptionalById(cabinToBeCreated.getSite().getId())
+				.orElseThrow(() -> new NotFoundException(ExceptionMessage.SITE_NOT_FOUND.getMessage()));
+		CabinEntity cabinEntity = cabinConverter.toEntity(cabinToBeCreated);
+		cabinEntity.setSite(site);
+		List<Integer> attributesId = cabinToBeCreated.getAttributes().stream().map(AttributeDto::getId)
+				.collect(Collectors.toList());
+		List<AttributeEntity> attributeEntities = new ArrayList<>();
```

Comment:
@fabiola0802 sugjerim per optimizim: ne vend qe te besh disa findById, mund te shtosh te repository i Attribute nje metode findByIds, ku i kalon nje liste me ids, dhe te kthen nje liste me attribute entities. ketu duhet perdorur `in` me criteriabuilder.
pastaj, filtron me AttributeType.CABIN, dhe merr listen e atyre qe kane vetem attribute type CABIN. Nje validim qe behet ketu eshte te kontrollosh nqs lista finale (ajo pasi behet `in` dhe pasi filtrohet me attribute type) ka te njejtin size me listen originale qe ka ids. nqs jo, hidhet nje validation exception (psh bad request). 
Ideja e kesaj, eshte qe te mos bejme 10 queries ne db per 10 attributes qe mund te kemi, por ti kalojme ne nje `in`, dhe te bejme vetem nje query.
Nje tjeter gje eshte qe ne vend te foreach, te perdoresh .filter me predicate dhe .collect(Collectors.toList()) ne fund

Reply:
meqe keto attributes perdoren dhe per site, kaloji me mire te AttributeService, dhe metoda te jete e tipit `List<AttributeEntity> findByIdsAndType(List<Integer> ids, AttributeType type)` dhe beji veprimet aty, dhe te create/update thjesht therrit kete si metode. Ne kete menyre ke dhe validimin e listes me id dhe per site, thjesht duke i kaluar attribute type